chain-extensions/pallet-scheduler/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+pub mod tests;
+pub mod weights;
+
+use frame_support::dispatch::Weight;
+use frame_support::traits::{schedule, Currency};
+use frame_system::RawOrigin;
+use pallet_contracts::{
+    chain_extension::{ChainExtension, Environment, Ext, InitState, RetVal, SysConfig},
+    Call as PalletContractCall,
+};
+use pallet_scheduler::WeightInfo;
+use parity_scale_codec::{Encode, HasCompact};
+use scale_info::TypeInfo;
+use scheduler_chain_extension_types::*;
+use sp_core::Get;
+use sp_runtime::traits::StaticLookup;
+use sp_runtime::DispatchError;
+use sp_std::boxed::Box;
+use sp_std::fmt::Debug;
+use sp_std::marker::PhantomData;
+
+enum SchedulerFunc {
+    Schedule,
+    Cancel,
+}
+
+impl TryFrom<u16> for SchedulerFunc {
+    type Error = DispatchError;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        match value {
+            1 => Ok(SchedulerFunc::Schedule),
+            2 => Ok(SchedulerFunc::Cancel),
+            _ => Err(DispatchError::Other(
+                "PalletSchedulerExtension: Unimplemented func_id",
+            )),
+        }
+    }
+}
+
+type BalanceOf<T> = <<T as pallet_contracts::Config>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+/// Pallet Scheduler chain extension.
+pub struct SchedulerExtension<T, W>(PhantomData<(T, W)>);
+
+impl<T, W> Default for SchedulerExtension<T, W> {
+    fn default() -> Self {
+        SchedulerExtension(PhantomData)
+    }
+}
```

Comment:
```suggestion
#[derive(DefaultNoBound)]
pub struct SchedulerExtension<T, W>(PhantomData<(T, W)>);
```

No need to manually impl `Default`, use `use frame_support::DefaultNoBound;`